cli/cook/subcommands/show.py

```diff
@@ -208,63 +217,44 @@ def satisfy_pred():
         resp_ = make_request_fn(cluster, uuids)
         return pred(resp_.json())
 
-    try:
-        resp = make_request_fn(cluster, uuids)
-        if resp.status_code == requests.codes.ok:
-            entities = resp.json()
-            if pred:
-                if entity_type == 'job':
-                    wait_text = 'Waiting for the following jobs'
-                elif entity_type == 'instance':
-                    wait_text = 'Waiting for instances of the following jobs'
-                elif entity_type == 'group':
-                    wait_text = 'Waiting for the following job groups'
-                else:
-                    raise Exception('Invalid entity type %s.' % entity_type)
-
-                uuid_text = ', '.join([e['uuid'] for e in entities])
-                wait_text = '%s on %s: %s' % (wait_text, colors.bold(cluster['name']), uuid_text)
-                index = progress.add(wait_text)
-                if pred(entities):
-                    progress.update(index, colors.bold('Done'))
-                else:
-                    entities = wait_until(satisfy_pred, timeout, interval)
-                    if entities:
-                        progress.update(index, colors.bold('Done'))
-                    else:
-                        raise Exception('Timeout waiting for response.')
-            return entities
+    entities = http.make_data_request(lambda: make_request_fn(cluster, uuids))
+    if pred and len(entities) > 0:
+        if entity_type == 'job':
+            wait_text = 'Waiting for the following jobs'
+        elif entity_type == 'instance':
+            wait_text = 'Waiting for instances of the following jobs'
+        elif entity_type == 'group':
+            wait_text = 'Waiting for the following job groups'
         else:
-            return {}
-    except requests.exceptions.ConnectionError as ce:
-        logging.info(ce)
-        return {}
-    except json.decoder.JSONDecodeError as jde:
-        logging.exception(jde)
-        return {}
-
+            raise Exception('Invalid entity type %s.' % entity_type)
 
-def make_entity_request(cluster, endpoint, params):
-    """Attempts to query entities corresponding to the given params from cluster."""
-    url = make_url(cluster, endpoint)
-    resp = http.get(url, params=params)
-    logging.info('response from cook: %s' % resp.text)
-    return resp
+        uuid_text = ', '.join([e['uuid'] for e in entities])
+        wait_text = '%s on %s: %s' % (wait_text, colors.bold(cluster['name']), uuid_text)
+        index = progress.add(wait_text)
+        if pred(entities):
+            progress.update(index, colors.bold('Done'))
+        else:
+            entities = wait_until(satisfy_pred, timeout, interval)
+            if entities:
+                progress.update(index, colors.bold('Done'))
+            else:
+                raise Exception('Timeout waiting for response.')
```

Comment:
Could we raise a more specific type of exception here? E.g., `TimeoutError` (built-in) or `requests.exceptions.Timeout`?

Reply:
Done